File: AlumniAPI_NET9/Models/Graduate.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Security.Policy;
using System.Text;
using System.Threading.Tasks;

namespace AlumniAPI.Models
{
    public class Graduate
    {
        public int Id { get; set; }

        public string FirstName { get; set; }
        public string LastName { get; set; }

        public int AcademicEntryYear { get; set; }
        public DateTime GraduationDate { get; set; }
        public double DegreeGrade { get; set; }

        public string Email { get; set; }
        public string PersonalWebsite { get; set; }

        public int UserId { get; set; }
        public virtual User? User { get; set; }

        public virtual ICollection<Phone> Phones { get; set; }
        public virtual ICollection<SocialMedia> SocialMedias { get; set; }
        public virtual ICollection<Employment> Employments { get; set; }

        public Graduate()
        {
            Phones = new List<Phone>();
            SocialMedias = new List<SocialMedia>();
            Employments = new List<Employment>();
        }
    }

}

File: alumni-tracker/src/Models/Graduate.ts
export interface Graduate {
  id: number;
  firstName: string;
  lastName: string;
  academicEntryYear: number;
  graduationDate: string; // ISO string from DateTime
  degreeGrade: number;
  email: string;
  personalWebsite: string;
  userId: number;
  phones: Phone[];
  socialMedias: SocialMedia[];
  employments: Employment[];
}

export interface Employment {
  id: number;
  organization: string;
  organizationSite: string;
  from: string; // ISO string
  to: string;   // ISO string
  jobType: string;
  jobDescription: string;
  relatedField: string;
  address: Address;
  graduateId: number;
}

export interface Address {
  street: string;
  number: string;
  city: string;
  postalCode: string;
  country: string;
  longitude: string;
  latitude: string;
}

export interface User {
  id: number;
  username: string;
  passwordHash: string;
  graduates: Graduate[];
}

export interface Phone {
  id: number;
  number: string;
  type: string;
  graduateId: number;
}

export interface SocialMedia {
  id: number;
  type: string;
  url: string;
  graduateId: number;
}